x-pack/plugins/maps/public/shared/layers/sources/wms_source/wms_create_source_editor.js

```diff
@@ -0,0 +1,223 @@
+  }
+
+  _handleStyleOptionsChange = (selectedOptions) => {
+    this.setState({
+      selectedStyleOptions: selectedOptions,
+      styles: selectedOptions.map(selectedOption => {
+        return selectedOption.value;
+      }).join(',')
+    }, this._previewIfPossible);
+  }
+
+  _renderLayerAndStyleInputs() {
+    if (!this.state.hasAttemptedToLoadCapabilities || this.state.isLoadingCapabilities) {
+      return null;
+    }
+
+    if (this.state.getCapabilitiesError || this.state.layerOptions.length === 0) {
+      return (
+        <Fragment>
+          <EuiCallOut
+            title={i18n.translate('xpack.maps.source.wms.getCapabilitiesErrorCalloutTitle', {
+              defaultMessage: 'Unable to load service metadata'
+            })}
+            color="warning"
+          >
+            <p>{this.state.getCapabilitiesError}</p>
+          </EuiCallOut>
+
+          <EuiFormRow
+            label={LAYERS_LABEL}
+            helpText={i18n.translate('xpack.maps.source.wms.layersHelpText', {
+              defaultMessage: 'use comma separated list of layer names'
+            })}
+          >
+            <EuiFieldText
+              onChange={this._handleLayersChange}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={STYLES_LABEL}
+            helpText={i18n.translate('xpack.maps.source.wms.stylesHelpText', {
+              defaultMessage: 'use comma separated list of style names'
+            })}
+          >
+            <EuiFieldText
+              onChange={this._handleStylesChange}
+            />
+          </EuiFormRow>
+        </Fragment>
+      );
+    }
+
+    return (
+      <Fragment>
+        <EuiFormRow
+          label={LAYERS_LABEL}
+        >
+          <EuiComboBox
+            options={this.state.layerOptions}
+            selectedOptions={this.state.selectedLayerOptions}
+            onChange={this._handleLayerOptionsChange}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={STYLES_LABEL}
+        >
+          <EuiComboBox
+            options={this.state.styleOptions}
+            selectedOptions={this.state.selectedStyleOptions}
+            onChange={this._handleStyleOptionsChange}
+          />
+        </EuiFormRow>
+      </Fragment>
+    );
+  }
+
+  render() {
+    return (
+      <EuiForm>
+        <EuiFormRow
+          label={i18n.translate('xpack.maps.source.wms.urlLabel', {
+            defaultMessage: 'Url'
+          })}
+        >
+          <EuiFieldText
+            value={this.state.serviceUrl}
+            onChange={this._handleServiceUrlChange}
+          />
+        </EuiFormRow>
+
+        <EuiButton
+          onClick={this._loadCapabilities}
+          isDisabled={!this.state.serviceUrl}
+          isLoading={this.state.isLoadingCapabilities}
+        >
+          <FormattedMessage
+            id="xpack.maps.source.wms.getCapabilitiesButtonText"
+            defaultMessage="Load capabilities"
+          />
+        </EuiButton>
```

Comment:
Maybe add some spacing (?) The button is squats on the form below

![image](https://user-images.githubusercontent.com/1833023/54131084-0234d000-43e8-11e9-9a56-84c0d2ba9e2b.png)